src/Input/index.jsx

```diff
@@ -0,0 +1,105 @@
+import React from 'react';
+import classNames from 'classnames';
+import PropTypes from 'prop-types';
+
+
+class Input extends React.Component {
+  componentDidMount() {
+    if (process.env.NODE_ENV === 'development') {
+      this.checkHasLabel();
+    }
+  }
+
+  getHTMLTagForType() {
+    const { type } = this.props;
+    if (type === 'select' || type === 'textarea') return type;
+    return 'input';
```

Comment:
Would it make sense here to whitelist the valid values?  I.e., I can totally see allowing a default if it's not specified, but maybe it's still worth checking for "penguin"?

Reply:
All set now in propTypes